### ISSUE TITLE
🚀 Feature: (GITEA) Functions Git Provider | Webhook Trigger

### ISSUE THREAD

**Issue by Schrolli91** [appwrite/appwrite#7121]
### 🔖 Feature description

I am using Gitea for self hosting my private projects and it would be nice if this will be integrated in the functions deployment mechanism too.

Maybe a good option is also, to trigger a deployment via an webhook?
So you can integrate every tool or script to trigger a deployment.
Maybe via an customizable deploy script (native git clone/install deps/etc...).

### 🎤 Pitch

Brings more integration into the available tool-sets out there.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
Thanks for raising this issue! 🙏🏼 We plan to support additional providers so it's good you created this feature request.

**Comment by Schrolli91** [comment]
Gitea also supports OAuth - maybe its possible to integrate some sort of a generic OAuth connection.

**Comment by stnguyen90** [comment]
@Schrolli91, unfortunately, not. There's much more to the GitHub integration than just authentication. For example, we make use of webhooks to automatically deploy to Appwrite when a branch is updated. In addition, comments are added to PRs for the progress of deployments.

**Comment by Schrolli91** [comment]
Is there some sort of simple and generic way planed?
Something like a webhook fired against appwrite and appwrite will clone a repo by itself, install dependencies, ...

**Comment by Shadowfita** [comment]
This feature would be very helpful.

**Comment by pixlguru** [comment]
Yes, please support gitea / forgejo integration, this would be awsome!

**Comment by Hanibachi** [comment]
Yes please

**Comment by barart** [comment]
+1

**Comment by kwazieharry** [comment]
+1

**Comment by Meldiron** [comment]
Gitea is now a supported provider after recent Appwrite Init, alongside GitLab, BitBucket 🥳
